crates/solidity/outputs/cargo/crate/src/backend/ast/generated/visitor.rs

```diff
@@ -0,0 +1,2701 @@
+        accept_identifier_path(&target.name, visitor);
+        if let Some(ref alias) = target.alias {
+            accept_using_alias(alias, visitor);
+        }
+    }
+    visitor.leave_using_deconstruction_symbol(target);
+}
+
+pub fn accept_using_alias(target: &UsingAlias, visitor: &mut dyn Visitor) {
+    if visitor.enter_using_alias(target) {
+        accept_using_operator(&target.operator, visitor);
+    }
+    visitor.leave_using_alias(target);
+}
+
+pub fn accept_contract_definition(target: &ContractDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_contract_definition(target) {
+        accept_contract_specifiers(&target.specifiers, visitor);
+        accept_contract_members(&target.members, visitor);
+    }
+    visitor.leave_contract_definition(target);
+}
+
+pub fn accept_inheritance_specifier(target: &InheritanceSpecifier, visitor: &mut dyn Visitor) {
+    if visitor.enter_inheritance_specifier(target) {
+        accept_inheritance_types(&target.types, visitor);
+    }
+    visitor.leave_inheritance_specifier(target);
+}
+
+pub fn accept_inheritance_type(target: &InheritanceType, visitor: &mut dyn Visitor) {
+    if visitor.enter_inheritance_type(target) {
+        accept_identifier_path(&target.type_name, visitor);
+        if let Some(ref arguments) = target.arguments {
+            accept_arguments_declaration(arguments, visitor);
+        }
+    }
+    visitor.leave_inheritance_type(target);
+}
+
+pub fn accept_storage_layout_specifier(target: &StorageLayoutSpecifier, visitor: &mut dyn Visitor) {
+    if visitor.enter_storage_layout_specifier(target) {
+        accept_expression(&target.expression, visitor);
+    }
+    visitor.leave_storage_layout_specifier(target);
+}
+
+pub fn accept_interface_definition(target: &InterfaceDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_interface_definition(target) {
+        if let Some(ref inheritance) = target.inheritance {
+            accept_inheritance_specifier(inheritance, visitor);
+        }
+        accept_interface_members(&target.members, visitor);
+    }
+    visitor.leave_interface_definition(target);
+}
+
+pub fn accept_library_definition(target: &LibraryDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_library_definition(target) {
+        accept_library_members(&target.members, visitor);
+    }
+    visitor.leave_library_definition(target);
+}
+
+pub fn accept_struct_definition(target: &StructDefinition, visitor: &mut dyn Visitor) {
+    if visitor.enter_struct_definition(target) {
+        accept_struct_members(&target.members, visitor);
+    }
+    visitor.leave_struct_definition(target);
+}
+
+pub fn accept_struct_member(target: &StructMember, visitor: &mut dyn Visitor) {
+    if visitor.enter_struct_member(target) {
+        accept_type_name(&target.type_name, visitor);
+    }
+    visitor.leave_struct_member(target);
+}
+
+pub fn accept_enum_definition(target: &EnumDefinition, visitor: &mut dyn Visitor) {
```

Comment:
WDYT of grouping the "acceptor" methods in a "receiver" struct? to organize imports and the file namespace?
i.e. something like this:

```rust
struct Receiver;

impl Receiver {
    fn receive_foo(visitor: &mut impl Visitor, source: &Foo) {
        // ....
    }
}
```

Reply:
I'm not sure about this. Right now, all acceptor methods (along with the `Visitor` trait) are namespaced under the `visitor` module, but there's nothing else in the module. What would be the advantage of grouping the acceptors in a different module?

Reply:
I think this could probably be discussed in the context of #1311